src/sonic-host-services/scripts/hostcfgd

```diff
@@ -530,6 +571,211 @@ class AaaCfg(object):
+
+            except subprocess.CalledProcessError as e:
+                syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(e.cmd, e.returncode, e.output))
+
+    def passwd_aging_warn_modify(self, curr_expiration_warning):
+        normal_accounts = self.get_normal_accounts()
+        if not normal_accounts:
+            syslog.syslog(syslog.LOG_ERR,"failed, no normal users found in /etc/passwd")
+            return
+
+        for normal_account in normal_accounts:
+
+            new_curr_expiration_warning = self.get_user_expire_time(normal_account, curr_expiration_warning)
+
+            try:
+                chage_p_w = subprocess.Popen(('chage', '-W ' + str(new_curr_expiration_warning), normal_account), stdout=subprocess.PIPE)
+                return_code_chage_p_w = chage_p_w.poll()
+                if return_code_chage_p_w != 0:
+                    syslog.syslog(syslog.LOG_ERR, "failed: return code - {}".format(return_code_chage_p_w))
+
+            except subprocess.CalledProcessError as e:
+                syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(e.cmd, e.returncode, e.output))
+
+    def get_user_expire_time(self, normal_account, curr_expiration):
+        last_passw_change = ''
+        new_curr_expiration = curr_expiration
+        REGEX_LAST_PASSW_CHANGE = r'^Last password change[ \t]*:[ \t]*(?P<last_passw_change>.*)'
+
+        try:
+            chage = subprocess.Popen(('chage', '-l', '-i', normal_account), stdout=subprocess.PIPE)
+            chage_data = chage.stdout.readlines()
+
+            for line in chage_data:
+                m1 = re.match(REGEX_LAST_PASSW_CHANGE, line.decode('utf-8'))
+                if m1:
+                    last_passw_change = m1.group("last_passw_change")
+                    break
+
+            # update the expire passw time according the last passw change
+            if not last_passw_change:
+                return curr_expiration
+
+            last_passw_change_obj = datetime.strptime(last_passw_change, '%Y-%m-%d')
+
+            today = datetime.now()
+            last_passw_change_diff = today - last_passw_change_obj
+            if (curr_expiration - last_passw_change_diff.days) > 0:
+                new_curr_expiration = curr_expiration - last_passw_change_diff.days
+            else:
+                new_curr_expiration = 0
+
+        except subprocess.CalledProcessError as err:
+            syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(err.cmd, err.returncode, err.output))
+
+        return new_curr_expiration
+
+    def is_passwd_aging_expire_update(self, curr_expiration):
+        """ Function verify that the current age expiry policy values are equal from the old one
+            Return update_age_status dict, the 'True' value meaning that was a modification from the last time.
```

Comment:
update_age_status is bool correct? Can you modify the comments?

Reply:
I will (old comment)